### PR TITLE
retry with exp backoff when rate limited by Amazon

### PR DESCRIPTION
**Pull request by stefanwire**
- retry with exp backoff when rate limited by Amazon
- add changelog

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/gundeck/src/Gundeck/Aws.hs [hunk 1]
```diff
@@ -236,7 +236,11 @@ updateEndpoint :: Set UserId -> Token -> EndpointArn -> Amazon ()
 updateEndpoint us tk arn = do
   let req = over SNS.setEndpointAttributes_attributes fun (SNS.newSetEndpointAttributes (toText arn))
   env <- ask
-  res <- retrying (limitRetries 1) (const isTimeout) (const (sendCatch (env ^. awsEnv) req))
+  res <-
+    retrying
+      (exponentialBackoff 50000 <> limitRetries 1)
+      (const $ pure . or . flip map [isTimeout, isRateLimited] . (&))
```
Comment (smatting) [inline services/gundeck/src/Gundeck/Aws.hs:242]:
```suggestion
      (const $ pure . or . flip map [isTimeout, isRateLimited] . (&))
```
Isn't there a more easy way to express this?

Comment (smatting) [reply services/gundeck/src/Gundeck/Aws.hs:242]: Also maybe a function can be abstracted for this to keep things more DRY

Comment (stefanwire) [reply services/gundeck/src/Gundeck/Aws.hs:242]: `const $ \x -> isTimeout x || isRateLimited x` or `const $ liftM2 (||) isTimeout isRateLimited`.

Comment (smatting) [reply services/gundeck/src/Gundeck/Aws.hs:242]: The first version looks easy to me

Comment (stefanwire) [reply services/gundeck/src/Gundeck/Aws.hs]: Both done.

**Review by smatting — APPROVED** [review]

**Review by smatting — APPROVED** [review]
If I understand this correctly all this PR does is add some exponential backing off. Numbers of retries stay the same

**Comment by stefanwire** [comment]
> If I understand this correctly all this PR does is add some exponential backing off. Numbers of retries stay the same

Yes, it also retries when receiving a 426 from Amazon. This is new as well.

**Pull request merged**